src/App.tsx

```diff
@@ -1,93 +1,91 @@
+      return {
+        ...todo,
+        completed: !isTodosCompleted,
+      };
+    }));
+  };
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+  const handleTodoUpdate = (todo: Todo, newValue: object) => {
+    const copyOfTodos = [...todos];
+    const updatedTodoIndex = todos.findIndex(({ id }) => id === todo.id);
 
-        <ul className="todo-list" data-cy="todoList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    copyOfTodos[updatedTodoIndex] = { ...todo, ...newValue };
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    setTodos(copyOfTodos);
+  };
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-editing" />
-              <label htmlFor="toggle-editing">zxcvbnm</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const handleQueryChange = (newQuery: string) => {
+    setQuery(newQuery);
+  };
 
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view2" />
-              <label htmlFor="toggle-view2">1234567890</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+  const handleTodoAdd = (todo: Todo) => {
+    setTodos([...todos, todo]);
+    setQuery('');
+  };
 
-      <footer className="footer">
-        <span className="todo-count" data-cy="todosCounter">
-          3 items left
-        </span>
+  const handleTodoDelete = (todoId: string) => {
+    setTodos(todos.filter(todo => todo.id !== todoId));
+  };
 
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
+  const handleClearCompleted = () => {
+    setTodos(todos.filter(({ completed }) => !completed));
+  };
 
-          <li>
-            <a href="#/active">Active</a>
-          </li>
+  const completedTodos = todos
```

Comment:
Minor: since, it's doesn't array of todos, better to name it in other way
```suggestion
  const completedTodosLength = todos
```